### PR TITLE
Fixes for /proc/self

### PR DESCRIPTION
**Pull request by bugaevc**
A couple of fixes in the kernel for resolving `/proc/self` and related paths.

Without this PR:
```
anon@courage:/home/anon$> ls -l /proc/self/exe
opendir: No such file or directory
Shell: ls(12) exited with status 1
```

With the first fix:
```
anon@courage:/home/anon$> ls -l /proc/self/exe
drwxrwxrwx  anon users          0   1985-02-09 02:23:00 vm/
drwxrwxrwx  anon users          0   1985-02-09 02:23:00 vmo/
drwxrwxrwx  anon users          0   1985-02-09 02:23:00 stack/
drwxrwxrwx  anon users          0   1985-02-09 02:23:00 regs/
drwxrwxrwx  anon users          0   1985-02-09 02:23:00 fds/
drwxrwxrwx  anon users          0   1985-02-09 02:23:00 exe/
drwxrwxrwx  anon users          0   1985-02-09 02:23:00 cwd/
drwxrwxrwx  anon users          0   1985-02-09 02:23:00 fd/
```
With both fixes, it works as expected:
```
anon@courage:/home/anon$> ls -l /proc/self/exe
drwxrwxrwx  anon users          0   1985-02-09 02:23:00 /proc/self/exe -> /bin/ls
```

I've also tried to handle (& tested) various other corner cases such as `/proc/self/cwd/../../////tmp///`, and it all seems to work as expected.

I've used this to debug the kernel, not sure why debugging is disabled upstream:
```patch
diff --git a/Makefile.common b/Makefile.common
index 60911ec7..7e80ee09 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -2,7 +2,7 @@ ARCH_FLAGS =
 STANDARD_FLAGS = -std=c++17 -Wno-sized-deallocation -fno-sized-deallocation
 WARNING_FLAGS = -Wextra -Wall -Wundef -Wcast-qual -Wwrite-strings -Wimplicit-fallthrough
 FLAVOR_FLAGS = -fno-exceptions -fno-rtti
-OPTIMIZATION_FLAGS = -Os
+OPTIMIZATION_FLAGS = -Os -g
 
 MAKEFILE_PATH := $(abspath $(lastword $(MAKEFILE_LIST)))
 SERENITY_BASE_DIR := $(patsubst %/,%,$(dir $(MAKEFILE_PATH)))
```

P.S. Thank you for such a great OS, @awesomekling! I've been watching your videos & following Serenity development for a while now, and finally got around to contributing.

### CODE REVIEW

**Comment by awesomekling** [comment]
Great fixes Sergey, I'm glad you finally got around to contributing! :)

I've always built without `-g` and I never felt like I was missing anything. Now I'm curious, what does it give you? I don't mind switching the default build over to `-g` unless there's some huge downside.

(Regardless, maybe we should have a way to build in debug or release mode with all the appropriate -g and -D flags.)

**Comment by bugaevc** [comment]
> I've always built without -g and I never felt like I was missing anything. Now I'm curious, what does it give you? I don't mind switching the default build over to -g unless there's some huge downside.

The ability to single-step in GDB, inspect variables, set breakpoints — all the usual debugging stuff.

**Comment by awesomekling** [comment]
Oh duh. I always forget that people use debuggers! :) That explains it.

I just tried doing a build with `-g` locally and it almost 10x'ed the size of every binary in the system. Let's keep the default build debug-free for now, but feel free to add any script that might make it easier for yourself to get the kind of build you want easily.
